src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+export const TodoModal = () => {
+  const dispatch = useDispatch();
+  const todo = useAppSelector(state => state.currentTodo);
 
-      <Loader />
+  const { id, title, completed, userId } = todo || {};
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    if (userId) {
+      setIsLoading(true);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+      getUser(userId)
+        .then(setUser)
+        .catch(error => {
+          throw new Error(error);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  }, [userId]);
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+  const handleCloseTodo = () => {
+    dispatch(setCurrentTodo(null));
+  };
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" />
+
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{id}
+            </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleCloseTodo}
+            />
+          </header>
+
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {title}
+            </p>
+
+            <p className="block" data-cy="modal-user">
+              {completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
+              {' by '}
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
You are using the `user` state variable directly in your JSX without checking if it's null. This could lead to a runtime error if the `user` is null. Consider adding a null check before accessing the `user` properties.